src/Api/PubnubApi/Entity/Subscription.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using PubnubApi.EventEngine.Common;
+
+namespace PubnubApi
+{
+	public class Subscription : SubscribeCapable
+	{
+		public override List<string> ChannelNames { get; set; } = new List<string>();
+		public override List<string> ChannelGroupNames { get; set; } = new List<string>();
+		public override Pubnub Pubnub { get; set; }
+		public override EventEmitter EventEmitter { get; set; }
+		public override SubscriptionOptions? Options { get; set; }
+		public override SubscribeCallbackExt Listener { get; set; } = new SubscribeCallbackExt();
+
+		public Subscription(string[] channels, string[] channelGroups, SubscriptionOptions? options, Pubnub pubnub, EventEmitter eventEmitter)
+		{
+			this.ChannelNames = channels.ToList();
+			this.ChannelGroupNames = channelGroups.ToList();
+			this.Pubnub = pubnub;
+			this.EventEmitter = eventEmitter;
+			this.Options = options;
+			this.EventEmitter.AddListener(Listener, channels, channelGroups);
+		}
+
+		public SubscriptionSet Add(Subscription subscription)
+		{
+			this.ChannelNames.AddRange(subscription.ChannelNames);
```

Comment:
Do we need a duplicate check here?

Reply:
The reason why we do not need duplicate check is:
If User has created subscription `S1` with `{c1, c2 }`. and subscription `S2` with `{c2, c3}`
now, If we Add only one entry of `c2` (which is overlapping channel).  And when user `Unsubscribe` `S2` then  at that moment we lose `c2` channel subscription. which impacts S1 subscription.

So solution is, We allow having duplicate entry for this.So that when one subscription calls unsubscribe then we remove one entry from duplicate entries. But when making final call to Subscribe API. We use `Distinct()` on channel/channelGroup array to optimise api call payload.  

Distinct() is called [here](https://github.com/pubnub/c-sharp/blob/CLEN-1851/src/Api/PubnubApi/EventEngine/Subscribe/SubscribeEventEngine.cs#L69) while making subscription call